### PR TITLE
Changelog update for 0.39

### PR DESCRIPTION
**Pull request by sklam**
as titled

### CODE REVIEW

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,3 +1,92 @@
+Version 0.39.0
+--------------
+
+Here are the highlights for the Numba 0.39.0 release.
+
+* This is the first version that supports Python 3.7.
+* With help from Intel, we have added fixed the SVML support (related issues
```
Comment (seibert) [inline CHANGE_LOG]: "With help from Intel, we have fixed some issues with SVML support"

Comment (stuartarchibald) [inline CHANGE_LOG]: s/added//

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,3 +1,92 @@
...
+  #2938, #2998, #3006).
+* List has gained support for containing reference-counted types like NumPy
+  arrays and `list`.  Note, list still can't hold heterogeneous types.
+* We have done a significant change to the internal calling-convention,
+  which should be transparent to most users, to allow for a future feature to
+  jump back into python-mode from a nopython-mode function.  This also fixed
```
Comment (stuartarchibald) [inline CHANGE_LOG]: s/fixed/fixes/

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,3 +1,92 @@
...
+  limitation to `print` that disables it usage from nopython functions that
```
Comment (stuartarchibald) [inline CHANGE_LOG]: s/limitation/a limitation/

and

s/it/its/

File: CHANGE_LOG [hunk 4]
```diff
@@ -1,3 +1,92 @@
...
+  are deep in the callstack.
+* For CUDA GPUs support, we added a `__cuda_array_interface__` following the
+  NumPy array interface specification to allow Numba to consume externally
+  defined device array.  We have opened a corresponding pull request to CuPy to
+  test out the concept and be able to use a CuPy GPU array.
```
Comment (stuartarchibald) [inline CHANGE_LOG]: Want to note the new HTML ipython/jupyter magic?

File: CHANGE_LOG [hunk 5]
```diff
@@ -1,3 +1,92 @@
...
+
+Here are the highlights for the Numba 0.39.0 release.
+
+* This is the first version that supports Python 3.7.
+* With help from Intel, we have added fixed the SVML support (related issues
+  #2938, #2998, #3006).
+* List has gained support for containing reference-counted types like NumPy
+  arrays and `list`.  Note, list still can't hold heterogeneous types.
+* We have done a significant change to the internal calling-convention,
+  which should be transparent to most users, to allow for a future feature to
```
Comment (stuartarchibald) [inline CHANGE_LOG]: s/to allow for a future feature to jump/to allow for a future feature that will permit jumping/ ?

File: CHANGE_LOG [hunk 6]
```diff
@@ -1,3 +1,92 @@
...
+* PR #2930: Add configuration file and color schemes.
+* PR #2928: Making error about untyped list more informative.
+* PR #2926: Enable fence for all architecture and add developer notes
+* PR #2902: Support for np.unique
+* PR #2840: Support list of refcounted types
+* PR #2793: Simplify and remove javascript from html_annotate templates.
+
+CUDA Enhancements:
+
+* PR #3037: Add CUDA IPC support on non-peer-accessible devices
```
Comment (stuartarchibald) [inline CHANGE_LOG]: Numerical {desc,asc}ending and ordered?

File: CHANGE_LOG [hunk 7]
```diff
@@ -1,3 +1,92 @@
...
+  limitation to `print` that disables it usage from nopython functions that
+  are deep in the callstack.
+* For CUDA GPUs support, we added a `__cuda_array_interface__` following the
+  NumPy array interface specification to allow Numba to consume externally
+  defined device array.  We have opened a corresponding pull request to CuPy to
+  test out the concept and be able to use a CuPy GPU array.
+
+Enhancements:
+
+* PR #3041: This adds @needs_blas to tests that use BLAS
```
Comment (stuartarchibald) [inline CHANGE_LOG]: descending order to be consistent with last 6ish releases? Probably doesn't matter :)

File: CHANGE_LOG [hunk 8]
```diff
@@ -1,3 +1,92 @@
...
+CUDA Enhancements:
+
+* PR #3037: Add CUDA IPC support on non-peer-accessible devices
+* PR #2860: __cuda_array_interface__
+* PR #2929: Add Flag To Prevent Unneccessary D->H Copies
+* PR #2910: More CUDA intrinsics
+
+CI Enhancements:
+
+* PR #3021: Update appveyor config.
```
Comment (stuartarchibald) [inline CHANGE_LOG]: Numerical {desc,asc}ending and ordered?

File: CHANGE_LOG [hunk 9]
```diff
@@ -1,3 +1,92 @@
...
+* PR #3042: Add catchsegv
+* PR #3040: Add fault handler to all builds
+
+Fixes:
+
+* PR #3048: [WIP] Fix cuda tests failure on buildfarm
```
Comment (stuartarchibald) [inline CHANGE_LOG]: Numerical {desc,asc}ending and ordered?

**Comment by stuartarchibald** [comment]
Merge approval over IRC from @sklam and @seibert. Thanks.

**Pull request merged**